docs/proposals/edge-authentication.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Edge Authentication Design
+authors:
+  - "@ls889"
+  - "@GsssC"
+  - "@XJangel"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@fisherxu"
+  - "@kadisi"
+
+creation-date: 2020-02-06
+last-updated: 2020-02-06
+status: alpha
+
+---
+
+# Edge Authentication Design
+
+## motivation
+
+The current connection between EdgeCore and CloudCore, that is, the authentication and authorization of the edge node to the cloud requires manual replication of the certificate, which has poor scalability and flexibility. And this root certificate is not within the K8S native authentication and authorization system and requires additional maintenance.
+
+## goal
+
+To implement that edge nodes automatically apply for a certificate when joining the cluster, and k8s api server approves the certificate.
+
+## design detail
+
+![token.png](../images/edgeAuthentication/edge_authentication.jpg)
+
+Update the tool keadm to simplify the complexity of joining edge nodes to the cluster.
+
+Firstly, Keadm establishes a TLS connection with Cloudcore. Then keadm sends a request of cluster-info to CLoudCore. Then the CloudCore will return the cluster-info containing  a public Key to keadm. Based on the discovery-token-ca-cert-hash flag, Keadm can validate the public key of the root certificate authority (CA) presented by the CloudCore. Then EdgeCore uses token to authenticate API Server,and CloudCore will forward token to the APIServer and forward a certificate to EdgeCore.If successful,EdgeCore and CloudCore will establish a wss connection.
+
+#### cloudcore：
+
+​	Cloudcore directly configures CA certificate and cloudcore certificate when accessing the api server. This cloudcore certificate is issued by a **same** CA with the edgecore certificate, so they can establish TLS.
+
+#### keadm join
+
+​	Authentication is split into discovery (having the Node trust the Kubernetes Master) and TLS bootstrap (having the Kubernetes Master trust the Node).
+
+##### 	preflight checks
+
+Check version of OS and install subsequently the required pre-requisites using supported steps. Currently we will support **ONLY** (Ubuntu & CentOS)
+
+Check and install all the pre-requisites before executing edge-controller, which are
+
+- Docker (currently 18.06.0ce3-0~ubuntu) and check is service is up.
+- mosquitto (latest available in OS repos) and check if running.
+
+##### 	discovery clustor-info
+
+​	`keadm join` is invoked with `--discovery-token`, token discovery is used; in this case the node basically retrieves the cluster CA certificates from the  `cluster-info` ConfigMap in the `kube-public` namespace.
```

Comment:
is it safe to put cluster CA certificates in cluster-info? What if the administrator want to stop the edge connection because of the resource or security reason?

Reply:
The cluster CA cert now has been put in  cluster-info, and cluster-info can be got without any authentication such as using a browser accessing "https://apiserver-ip:6443/api/v1/namespaces/kube-public/configmaps/cluster-info". For the second, now we only consider the case where nodes join the cluster，if edgecore issue a illegal request, I think its no need to disconnect the edgecore and just return access error.